### DISCUSSION TITLE
Debian Sid, sdrplay API 3.12 vs 3.14

### DISCUSSION THREAD

**Discussion by kobusjooste** [AlexandreRouma/SDRPlusPlus#1312] [General]
Hi I'm 99% sure I'm following the build instructions correctly.

I'm on Debian SID and I have all the required libraries and dev packages installed. I'm able to successfully cmake, make and make install.

My problem is with the SDRplay module. Upon running I get:

[ERROR] Couldn't load /usr/lib/sdrpp/plugins/sdrplay_source.so: libsdrplay_api.so.3.12: cannot open shared object file: No such file or directory

I have API version 3.14 installed, but sdr++ insists on requiring 3.12. I can't find where this is hardcoded in the code. 

I have even tried to fool it with symlinks but nothing works. (symlinking 3.12 to the 3.14) 

FYI I have SDRAngel and a number of other packages correctly compiled from source and installed with SDRPlay support so I'm inclined to believe that my system is set up correctly.

Any ideas on how to get SDRPlusPlus working with the newest SDRPlay API version 3.14?

Thanks,
Koos

**Comment by AlexandreRouma** [comment]
> FYI I have SDRAngel and a number of other packages correctly compiled from source

Well then compile SDR++ from source as well?
The issue is simply that the nightly are compiled for 3.12, if you're not running 3.12 you have to build from source so the versions match.

**Reply by AlexandreRouma** [reply]
You still have 3.12 installed then. The linker isn't gonna magically make up an API version, none of it is hardcoded in the source.
And the nightly do have SDRplay support.

**Reply by kobusjooste** [reply]
Hi Alexandre

I managed to solve my problems

1. a big part of the problem was that the nightly was still installed. you are right the nightly requires SDRPlay API 3.12 - this was indeed the errors that i was seeing.

2. further i moved my install path to /usr/local. i think another part of my problem was installing into a nonstandard place in /opt

thanks for your time and for development on SDR++

Regards.
